### ISSUE TITLE
Type inference for .toObject

### ISSUE THREAD

**Issue by DrecDroid** [mihaifm/linq#88]
## From:
```typescript
toObject(keySelector: (element: T) => any, elementSelector?: (element: T) => any): Object;
```

## To:
```typescript
toObject<TKey, TElement>(keySelector: (element: T) => TKey, elementSelector?: (element: T) => TElement): Record<TKey, TElement>;
```

**Comment by mihaifm** [comment]
Thanks, I will fix this in the next version.

**Comment by mihaifm** [comment]
Fixed in version 4.0.0. Thanks for the suggestion.
